### PR TITLE
[Merged by Bors] - feat(CategoryTheory): sheafification over essentially small sites

### PR DESCRIPTION
**Pull request by dagurtomas**
This PR provides the necessary API to sheafify over large but essentially small categories. 

- Given an equivalence of categories and a Grothendieck topology on one of them, we transport the topology along the equivalence and prove that the sheaf categories are equivalent.
- We prove that the property `HasSheafify` transports over the equivalence of sheaf categories mentioned above, and same for the property `HasSheafCompose` 
- We also add instances for the existence of small limits on sheaf categories on essentially small sites.

- [x] depends on: #9012 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Sites/Equivalence.lean [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    have : (Sieve.coverByImage e.inverse U).arrows g := Presieve.in_coverByImage _ g
+    replace := Sieve.downward_closed _ this (e.unit.app Y)
+    simpa using this
+
+instance : IsContinuous e.inverse (e.locallyCoverDense J).inducedTopology J :=
+  IsCoverDense.isContinuous _ _ _ (e.locallyCoverDense J).inducedTopology_coverPreserving
+
+/-- The functor in the equivalence of sheaf categories. -/
+@[simps!]
+def sheafCongr_functor : Sheaf J A ⥤ Sheaf (e.locallyCoverDense J).inducedTopology A where
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Sites/Equivalence.lean:112]: In this situation, we should use a name like `sheafCongrFunctor` or, better, `sheafCongr.functor`, for the auxiliary definitions.

File: Mathlib/CategoryTheory/Sites/Equivalence.lean [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+      apply HasSheafCompose.isSheaf
+      exact e.inverse.op_comp_isSheaf K J ⟨P, hP⟩
+    replace hP' : Presheaf.IsSheaf J (e.functor.op ⋙ e.inverse.op ⋙ P ⋙ F) :=
+      e.functor.op_comp_isSheaf _ _ ⟨_, hP'⟩
+    exact (Presheaf.isSheaf_of_iso_iff ((isoWhiskerRight e.op.unitIso.symm (P ⋙ F)))).mp hP'
+
+end Equivalence
+
+variable {C : Type (u+1)} [LargeCategory C] [EssentiallySmall C] (J : GrothendieckTopology C)
+variable (A : Type*) [Category A] --[LargeCategory A]
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Sites/Equivalence.lean:198]: The following seems more general to me, and it seems it compiles:
```suggestion
variable {C : Type*} [Category C] [EssentiallySmall C] (J : GrothendieckTopology C)
variable (A : Type*) [Category A]
```
Would you agree? (And does this work for your application?)

**Review by joelriou — APPROVED** [review]
Up to minor suggestions, this looks very good to me!

**Comment by dagurtomas** [comment]
Thanks! I applied both your suggestions

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**